### PR TITLE
Feature/km/a11y focus management

### PR DESCRIPTION
**Pull request by resource11**
Accessibility enhancements including:
1. Improved sidebar button toggling
2. Sidebar links keyboard focusable
3. Improved focus styling
4. Adds aria attributes
5. Sets focus to updated iframe page on BookItem click
6. Refactors sidebar for animation and screenreader support
7. Refactors css for efficiency/adds theming
8. Lints code

### CODE REVIEW

**Comment by resource11** [comment]
@cjshawMIT, can we walk through these hound errors when you get back to the office? Also, should we have Travis CI set up for this repo?

**Comment by coleshaw** [comment]
@resource11 Yes, Travis would be a good idea. I'll enable it in Travis, can you create / include the `.travis.yml` file in this PR? You should be able to copy one from another project.

**Comment by coleshaw** [comment]
@resource11 Looks like you also need to copy the file `client/ci.karma.conf.js` from OEA, to make the Travis tests work.

**Comment by resource11** [comment]
Yes, that's the next on the list... thanks for the reminder!

File: client/js/components/chrome/bookItem.jsx [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
   }
 
-  return <div className={bookItemClass} onClick={()=>openPage()}>
-    <a>
-      {props.content.navLabel}
-    </a>
-  </div>;
+  return (
+    <li className={bookItemClass} >
+      <a
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/components/chrome/bookItem.jsx:20]: fixed. Aded uri hash

File: client/js/components/book/page.jsx [hunk 2]
```diff
@@ -1,72 +1,113 @@
...
         this.scrollToAssessment();
         break;
+      default:
+        break;
     }
   }
 
-  componentDidMount() {
-    window.addEventListener("message", (e) => this.onMessage(e), false);
+  scrollToAssessment() {
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/components/book/page.jsx]: I'm stumped on this one.

File: client/js/libs/parser.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // flattens an object (recursively!), similarly to Array#flatten
 // e.g. flatten({ a: { b: { c: "hello!" } } }); // => "hello!"
 function flatten(object) {
-  var check = _.isPlainObject(object) && _.size(object) === 1;
+  const check = _.isPlainObject(object) && _.size(object) === 1;
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/libs/parser.js]: Touching parser.js at all breaks content player. We should ignore linting this one.

File: client/js/middleware/epub.spec.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ const mocks = {
   fakeRequest: () => {
     requestCalled = true;
     return new Promise((resolve, reject) => {
-      resolve({text:""});
+      resolve({text:''});
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/middleware/epub.spec.js]: touching epub.js and epub.spex.js breaks content player. We should ignore linting this one.

File: client/js/middleware/unplatform_session_management.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
-import api         from "../libs/api";
-import Network     from '../constants/network';
-import { DONE }    from "../constants/wrapper";
-import { browserHistory } from 'react-router';
+import api                      from '../libs/api';
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/middleware/unplatform_session_management.js]: Not touching this one...

File: client/specs_support/rapper.jsx [hunk 6]
```diff
@@ -1,12 +1,9 @@
...
 import React                    from 'react';
-import { connect }              from 'react-redux';
 
-export default class Rapper extends React.Component{
+export default class Rapper extends React.Component {
 
-  render(){
-    if(this.props.child) return <div><this.props.child ref="original" {...this.props.childProps}/></div>;
+  render() {
+    if (this.props.child) return <div><this.props.child ref="original" {...this.props.childProps} /></div>;
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/specs_support/rapper.jsx]: same here... not touching.

File: client/specs_support/stub.jsx [hunk 7]
```diff
@@ -1,10 +1,8 @@
-"use strict";
-
 import React from 'react';
 
-export default class Stub extends React.Component{
+export default class Stub extends React.Component {
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/specs_support/stub.jsx]: Ditto...

File: client/js/components/chrome/sidebar.jsx [hunk 8]
```diff
@@ -27,41 +31,73 @@ export class Sidebar extends React.Component{
...
+    let tableOfContents = '';
 
     if(this.props.sidebarOpen){
-      var tableOfContents = this.tableOfContents(this.props);
+      btnToggleClass = 'c-sidebar__toggle-button c-sidebar__toggle-button--open';
+      svgSpinClass = 'c-sidebar__svg c-sidebar__svg--spin';
+      btnAriaPressed = 'true';
+      btnAriaExpanded = 'true';
+      sidebarClass = 'c-sidebar c-sidebar--open';
+      unit = this.props.tocMeta.gradeUnit;
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/components/chrome/sidebar.jsx:65]: it's in there now, hound not recognizing?

File: client/js/components/chrome/sidebar.jsx [hunk 9]
```diff
@@ -27,41 +31,73 @@ export class Sidebar extends React.Component{
...
+      subject = this.props.tocMeta.subjectLesson;
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/components/chrome/sidebar.jsx:66]: it's in there now, hound not recognizing?

File: client/js/components/book/page.jsx [hunk 10]
```diff
@@ -198,30 +223,33 @@ export class Page extends React.Component {
...
-    return <iframe
-      onLoad={() => this.addIframeEventListeners()}
-      ref={(iframe) => this.contentIframe = iframe }
-      src={`${props.contentPath}/${current.content}`}
-      title={iframeTitle}
-      allowFullScreen="true" />;
+    if (!current) { return; }
+
+    const iframeTitle = `${this.props.tocMeta.gradeUnit} ${this.props.tocMeta.subjectLesson}`;
+    return (
```
[the remark this answers is not in the document]
Comment (resource11) [reply client/js/components/book/page.jsx]: holding for now. Tests passing, app builds successfully.

**Comment by coleshaw** [comment]
@resource11 please pull / update locally and review this. I just changed `yarn test` to `npm test` per this (perhaps related?) [issue on the Ember code base](https://github.com/ember-cli/ember-cli/issues/7328).

**Review by coleshaw — APPROVED** [review]
LGTM

**Pull request merged**